### ISSUE TITLE
[Bug] PCL2 cannot detect mod magiclib, version 0.7.329+ed83ba7-beta and magiclib-1.19.4-0.7.330+0499b91-beta

### ISSUE THREAD

**Issue by FourFeaves** [Hendrix-Shen/MagicLib#32]
### Minecraft Version

1.19.4

### MagicLib Version

0.7.329+ed83ba7-beta and magiclib-1.19.4-0.7.330+0499b91-beta

### Description

For minecraft version 1.19.4, carpet version  fabric-carpet-1.19.4-1.4.101+v230319 and fabric api version 0.82.0+1.19.4.
PCL2 launcher could not find the magiclib mod.

### To Reproduce

For minecraft version 1.19.4, carpet version  fabric-carpet-1.19.4-1.4.101+v230319 and fabric api version 0.82.0+1.19.4.
PCL2 launcher could not find the magiclib mod and crashed when the starup.

### Log

[游戏崩溃前的输出.txt](https://github.com/Hendrix-Shen/MagicLib/files/11581063/default.txt)

### Latest CI build tested

- [X] I confirm that this problem still exists with the latest CI build.

**Comment by FourFeaves** [comment]
```
[13:37:27] [main/INFO]: Loading Minecraft 1.19.4 with Fabric Loader 0.14.21
[13:37:27] [ForkJoinPool-1-worker-4/WARN]: Mod imblocker uses the version 1.0.20_pre5 which isn't compatible with Loader's extended semantic version format (Could not parse version number component '20_pre5'!), SemVer is recommended for reliably evaluating dependencies and prioritizing newer version
[13:37:27] [main/WARN]: Mod resolution failed
[13:37:27] [main/INFO]: Immediate reason: [HARD_DEP_NO_CANDIDATE masa_gadget_mod 3.1.289+1ca4352-stable {depends magiclib @ [>=0.7.293]}, ROOT_FORCELOAD_SINGLE masa_gadget_mod 3.1.289+1ca4352-stable]
[13:37:27] [main/INFO]: Reason: [HARD_DEP masa_gadget_mod 3.1.289+1ca4352-stable {depends magiclib @ [>=0.7.293]}, HARD_DEP ommc 0.5 {depends magiclib @ [>=0.5.30]}, HARD_DEP tweakmyclient 3.6.395+901cd91-beta {depends magiclib @ [>=0.7.329+ed83ba7-beta]}]
[13:37:27] [main/INFO]: Fix: add [add:magiclib 0.7.329+ed83ba7-beta ([[0.7.329+ed83ba7-beta,∞)])], remove [], replace []
[13:37:27] [main/ERROR]: Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Install magiclib, version 0.7.329+ed83ba7-beta or later.
Unmet dependency listing:
	 - Mod 'Masa Gadget' (masa_gadget_mod) 3.1.289+1ca4352-stable requires version 0.7.293 or later of magiclib, which is missing!
	 - Mod 'Oh My Minecraft Client' (ommc) 0.5 requires version 0.5.30 or later of magiclib, which is missing!
	 - Mod 'Tweak My Client' (tweakmyclient) 3.6.395+901cd91-beta requires version 0.7.329+ed83ba7-beta or later of magiclib, which is missing!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51) ~[fabric-loader-0.14.21.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192) ~[fabric-loader-0.14.21.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148) ~[fabric-loader-0.14.21.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) ~[fabric-loader-0.14.21.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.21.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at oolloo.jlw.Wrapper.invokeMain(Wrapper.java:60) ~[?:?]
	at oolloo.jlw.Wrapper.main(Wrapper.java:53) ~[?:?]
```

**Comment by Hendrix-Shen** [comment]
See #7.

**Comment by FourFeaves** [comment]
这是个好问题，但是我没看懂

麻将今天倒闭了吗
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Hendrix-Shen/MagicLib"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月27日(星期六) 中午1:39
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Hendrix-Shen/MagicLib] [Bug] PCL2 cannot detect mod magiclib, version 0.7.329+ed83ba7-beta and magiclib-1.19.4-0.7.330+0499b91-beta (Issue #32)





 
Closed #32 as completed.
